subsys/event_manager/event_manager.c

```diff
@@ -30,34 +31,42 @@ static bool log_is_event_displayed(const struct event_type *et)
 	return atomic_test_bit(_event_manager_event_display_bm.flags, idx);
 }
 
+static void log_event_using_buffer(const struct event_header *eh, const struct event_type *et)
+{
+	char log_buf[CONFIG_EVENT_MANAGER_EVENT_LOG_BUF_LEN];
+
+	int pos = et->log_event(eh, log_buf, sizeof(log_buf));
+
+	if (pos < 0) {
+		log_buf[0] = '\0';
+	} else if (pos >= sizeof(log_buf)) {
+		log_buf[sizeof(log_buf) - 2] = '~';
+	}
+
+	if (IS_ENABLED(CONFIG_EVENT_MANAGER_LOG_EVENT_TYPE)) {
+		LOG_INF("e: %s %s", et->name, log_strdup(log_buf));
+	} else {
+		LOG_INF("%s", log_strdup(log_buf));
+	}
+
+}
+
 static void log_event(const struct event_header *eh)
 {
 	const struct event_type *et = eh->type_id;
+	BUILD_ASSERT(CONFIG_EVENT_MANAGER_EVENT_LOG_BUF_LEN != 1);
```

Comment:
Maybe move under `log_event_using_buffer`? There you could quickly see why you need the build assert.

Reply:
If you want you can even add a comment to explain it.

Reply:
definitely move it to log_event_using_buffer - also change the value